python/rmm/rmm.py

```diff
@@ -233,6 +138,78 @@ def get_info(stream=0):
     return librmm.rmm_getinfo(stream)
 
 
+try:
+    import numba
+except Exception:
+    numba = None
+    _numba_memory_manager = None
+
+if numba:
+    class RMMNumbaManager(HostOnlyCUDAMemoryManager):
+        def __init__(self, logging=False):
+            super().__init__()
+            self._initialized = False
+            self._logging = logging
+
+        def memalloc(self, nbytes, stream=0):
+            buf = librmm.DeviceBuffer(size=nbytes, stream=stream)
+            ctx = cuda.current_context()
+            ptr = ctypes.c_uint64(int(buf.ptr))
+            mem = MemoryPointer(ctx, ptr, nbytes, owner=buf)
+            return mem
+
+        def get_ipc_handle(self, memory, stream=0):
+            """
+            Get an IPC handle from the DeviceArray ary with offset modified by
+            the RMM memory pool.
+            """
+            # Not a very clean implementation - may want to implement something
+            # at the C++ layer for this, and also not rely on borrowing bits of
+            # Numba internals to initialise ipchandle.
+            ipchandle = (ctypes.c_byte * 64)()  # IPC handle is 64 bytes
+            cuda.cudadrv.memory.driver_funcs.cuIpcGetMemHandle(
+                ctypes.byref(ipchandle),
+                memory.owner.handle,
+            )
+            source_info = cuda.current_context().device.get_device_identity()
+            ptr = memory.device_ctypes_pointer.value
+            offset = librmm.rmm_getallocationoffset(ptr, stream)
+            from numba.cuda.cudadrv.driver import IpcHandle
+            return IpcHandle(memory, ipchandle, memory.size, source_info,
+                             offset=offset)
+
+        def get_memory_info(self):
+            return get_info()
+
+        def initialize(self):
+            super().initialize()
+            if not self._initialized:
+                reinitialize(logging=self._logging)
```

Comment:
How can we pass additional kwargs here? I.E. if we want to use a pool or managed memory?

Reply:
I realized it was a mistake to be calling `reinitialize()` in the `reset()` function anyway - `reset` should only clear all allocations from the current context, not reset the state of the entire memory manager.

If you do want to call `reinitialize()`, I'd expect that it would be done outside of the Numba EMM plugin interface. For example, at program startup you might:

1. Call `reinitialize` with approprate parameters for a pool, or managed memory.
2. Set the Numba memory manager (e.g. calling `rmm.use_rmm_for_numba()`.
3. Start using Numba (allocating device arrays, calling kernels, etc.)

Reinitializing RMM in the middle of program execution (if you wanted different settings) should also work OK, as long as you have reset any contexts that you were using first (so that you don't have allocations from before re-initialization lying around).

I'll push a commit shortly which removes this call to `reinitialize` from `reset`.